docs/resilience.md

```diff
@@ -0,0 +1,144 @@
+# Resilience
+
+This document aims to provide a convenient overview of all resilience patterns that are supported by Riptide:
+
+## Bulkheads
+
+> Isolate elements of an application into pools so that if one fails, the others will continue to function.
+> This pattern is named Bulkhead because it resembles the sectioned partitions of a ship's hull. If the hull of a ship is compromised, only the damaged section fills with water, which prevents the ship from sinking.
+> 
+> [Microsoft: Bulkhead pattern](https://docs.microsoft.com/en-us/azure/architecture/patterns/bulkhead)
+
+Riptide supports the bulkhead pattern by supporting isolated, fixed-size thread pools and bounded work queues per 
+client/instance. 
+
+```yaml
+riptide.clients:
```

Comment:
I found the [complete example](https://github.com/zalando/riptide/tree/master/riptide-spring-boot-starter#configuration) helpful - maybe useful to link to it?